src/popup/router/pages/SignAndVerifyMsg.vue

```diff
@@ -0,0 +1,196 @@
+            <p :class="alert.class"><small>{{alert.msg}}</small></p>
+            <ae-button face="round" fill="primary" extend :class="[ signMessage.length > 0 ? '' : 'disabled' ]" @click="signMessageAction">Sign</ae-button>
+        </div>
+        <div v-if="page=='verify'">
+            <div class="actions">
+                <button class="backbutton toAccount" @click="page = ''"><ae-icon name="back" />{{ language.buttons.backToSignVerifyMsg }}</button>
+            </div>
+            <h4>Verify message</h4>
+            <hr>
+            <ae-textarea v-model="verifyMessage" monospace placeholder="Message" />
+            <p :class="alert.class"><small>{{alert.msg}}</small></p>
+            <ae-button face="round" fill="primary" extend :class="[ verifyMessage.length > 0 ? '' : 'disabled' ]" @click="verifyMessageAction">Verify</ae-button>
+        </div>
+        <popup :popupSecondBtnClick="popup.secondBtnClick"></popup>
+        <Loader size="small" :loading="loading" v-bind="{'content':''}"></Loader>
+    </div>
+</template>
+
+<script>
+import locales from '../../locales/locales.json';
+import { mapGetters } from 'vuex';
+import { Crypto } from '@aeternity/aepp-sdk/es';
+import Ae from '@aeternity/aepp-sdk/es/ae/universal';
+import { decrypt, str2buf } from '../../utils/keystore';
+
+export default {
+    data() {
+        return {
+            language: locales['en'],
+            loading: false,
+            page: '',
+            signMessage: '',
+            verifyMessage: '',
+            modalVisible: '',
+            signature: '',
+            alert: {
+                msg: '',
+                class: '',
+            },
+        }
+    },
+    locales,
+    computed: {
+        ...mapGetters(['account', 'sdk', 'activeAccount', 'popup', 'wallet']),
+    },
+    created() {
+        // console.log(this.account.encryptedPrivateKey);
+    },
+    methods:{
+        signMessageAction() {
+                this.loading = true;
+                browser.storage.sync.get('userAccount').then(async (user) => {
+                    if(user.userAccount && user.hasOwnProperty('userAccount')) {
+                        let encPrivateKey = user.userAccount.encryptedPrivateKey;
+                        try {
+                            JSON.parse(user.userAccount.encryptedPrivateKey);
+                        }catch(e) {
+                            user.userAccount.encryptedPrivateKey = JSON.stringify( user.userAccount.encryptedPrivateKey );
+                            encPrivateKey = JSON.stringify( user.userAccount.encryptedPrivateKey );
+                        }
+                        let encryptedPrivateKey = JSON.parse(user.userAccount.encryptedPrivateKey);
+                        let privKey = await decrypt(encryptedPrivateKey.crypto.ciphertext,'123123123',encryptedPrivateKey.crypto.cipher_params.nonce,encryptedPrivateKey.crypto.kdf_params.salt);
+                        let privKey64 = str2buf(privKey);
```

Comment:
You can use `Buffer.from(privKey, 'hex')` instead.

Reply:
The signature response is:
` { "type": "Buffer", "data": [ 226, 239, 148, 39, 119, 148, 76, 69, 16, 14, 130, 48, 41, 81, 195, 155, 193, 148, 206, 193, 244, 25, 148, 71, 232, 71, 97, 64, 37, 149, 97, 161, 101, 37, 73, 5, 103, 24, 61, 219, 207, 169, 59, 1, 50, 10, 11, 108, 27, 53, 39, 138, 20, 172, 62, 164, 50, 96, 200, 167, 54, 203, 123, 1 ] }`
And the Verify function again returns false.

Reply:
You can check the integration tests in the js sdk here for a stepping stone: https://github.com/aeternity/aepp-sdk-js/blob/e1b405e4dcf45b55ea91f1a84d3698d15eb8bc34/test/unit/crypto.js#L96

Reply:
Already tried, but again returns false.
I am 90% sure that i pass the right parameters.

Reply:
This could be related to https://github.com/waellet/waellet/issues/117.

I'll investigate this now.